cli/azd/internal/appdetect/spring_boot.go

```diff
@@ -0,0 +1,363 @@
+		logServiceAddedAccordingToMavenDependency(newDep.ResourceDisplay(), targetGroupId, targetArtifactId)
+		for bindingName, destination := range bindingDestinations {
+			log.Printf("  Detected Kafka Topic [%s] for binding [%s] by analyzing property file.",
+				destination, bindingName)
+		}
+	}
+}
+
+func detectStorageAccount(azdProject *Project, springBootProject *SpringBootProject) {
+	detectStorageAccountAccordingToSpringCloudStreamBinderMavenDependencyAndProperty(azdProject, springBootProject)
+}
+
+func detectStorageAccountAccordingToSpringCloudStreamBinderMavenDependencyAndProperty(
+	azdProject *Project, springBootProject *SpringBootProject) {
+	var targetGroupId = "com.azure.spring"
+	var targetArtifactId = "spring-cloud-azure-stream-binder-eventhubs"
+	var targetPropertyName = "spring.cloud.azure.eventhubs.processor.checkpoint-store.container-name"
+	if hasDependency(springBootProject, targetGroupId, targetArtifactId) {
+		bindingDestinations := getBindingDestinationMap(springBootProject.applicationProperties)
+		containsInBindingName := ""
+		for bindingName := range bindingDestinations {
+			if strings.Contains(bindingName, "-in-") { // Example: consume-in-0
+				containsInBindingName = bindingName
+				break
+			}
+		}
+		if containsInBindingName != "" {
+			targetPropertyValue := springBootProject.applicationProperties[targetPropertyName]
+			newDep := AzureDepStorageAccount{
+				ContainerNames: []string{targetPropertyValue},
+			}
+			azdProject.AzureDeps = append(azdProject.AzureDeps, newDep)
+			logServiceAddedAccordingToMavenDependencyAndExtraCondition(newDep.ResourceDisplay(), targetGroupId,
+				targetArtifactId, "binding name ["+containsInBindingName+"] contains '-in-'")
+			log.Printf("  Detected Storage Account container name: [%s] by analyzing property file.",
+				targetPropertyValue)
+		}
+	}
+}
+
+func detectSpringCloudAzure(azdProject *Project, springBootProject *SpringBootProject) {
+	var targetGroupId = "com.azure.spring"
+	var targetArtifactId = "spring-cloud-azure-starter"
+	if hasDependency(springBootProject, targetGroupId, targetArtifactId) {
+		newDep := SpringCloudAzureDep{}
+		azdProject.AzureDeps = append(azdProject.AzureDeps, newDep)
+		logServiceAddedAccordingToMavenDependency(newDep.ResourceDisplay(), targetGroupId, targetArtifactId)
+	}
+}
+
+func logServiceAddedAccordingToMavenDependency(resourceName, groupId string, artifactId string) {
+	logServiceAddedAccordingToMavenDependencyAndExtraCondition(resourceName, groupId, artifactId, "")
+}
+
+func logServiceAddedAccordingToMavenDependencyAndExtraCondition(
+	resourceName, groupId string, artifactId string, extraCondition string) {
+	insertedString := ""
+	extraCondition = strings.TrimSpace(extraCondition)
+	if extraCondition != "" {
+		insertedString = " and " + extraCondition
+	}
+	log.Printf("Detected '%s' because found dependency '%s:%s' in pom.xml file%s.",
+		resourceName, groupId, artifactId, insertedString)
+}
```

Comment:
Use one entry to print related log, which can keep all related log have same format.